src/Discord.Net.Rest/Entities/Channels/RestTextChannel.cs

```diff
@@ -168,5 +173,13 @@ IAsyncEnumerable<IReadOnlyCollection<IUser>> IChannel.GetUsersAsync(CacheMode mo
             else
                 return AsyncEnumerable.Empty<IReadOnlyCollection<IGuildUser>>();
         }
+
+        // INestedChannel
+        async Task<ICategoryChannel> INestedChannel.GetCategoryAsync()
```

Comment:
I think we need to change this signature (add CacheMode and RequestOptions) to follow the same pattern as other rest entities and call the inner `GetCategoryAsync`.

See:
https://github.com/RogueException/Discord.Net/blob/dev/src/Discord.Net.Rest/Entities/Guilds/RestGuild.cs#L295
https://github.com/RogueException/Discord.Net/blob/dev/src/Discord.Net.Rest/Entities/Channels/RestTextChannel.cs#L141